File: src/main/java/com/smarthome/backend/model/devices/helper/DeviceListenerParams.java
package com.smarthome.backend.model.devices.helper;

/**
 * Parameter-Klasse für Listener-Registrierung bei Device-Klassen.
 */
public class DeviceListenerParams {
    private String key;
    private String name;
    private Object param1;
    private Object param2;
    
    public DeviceListenerParams() {
    }
    
    public DeviceListenerParams(String key, String name) {
        this.key = key;
        this.name = name;
    }
    
    public DeviceListenerParams(String key, String name, Object param1) {
        this.key = key;
        this.name = name;
        this.param1 = param1;
    }
    
    public DeviceListenerParams(String key, String name, Object param1, Object param2) {
        this.key = key;
        this.name = name;
        this.param1 = param1;
        this.param2 = param2;
    }
    
    public String getKey() {
        return key;
    }
    
    public void setKey(String key) {
        this.key = key;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public Object getParam1() {
        return param1;
    }
    
    public void setParam1(Object param1) {
        this.param1 = param1;
    }
    
    public Object getParam2() {
        return param2;
    }
    
    public void setParam2(Object param2) {
        this.param2 = param2;
    }
    
    public Integer getParam1AsInt() {
        if (param1 instanceof Integer) {
            return (Integer) param1;
        } else if (param1 instanceof Number) {
            return ((Number) param1).intValue();
        }
        return null;
    }
    
    public String getParam1AsString() {
        return param1 != null ? param1.toString() : null;
    }
    
    public Boolean getParam1AsBoolean() {
        if (param1 instanceof Boolean) {
            return (Boolean) param1;
        }
        return null;
    }
    
    public Integer getParam2AsInt() {
        if (param2 instanceof Integer) {
            return (Integer) param2;
        } else if (param2 instanceof Number) {
            return ((Number) param2).intValue();
        }
        return null;
    }
    
    public String getParam2AsString() {
        return param2 != null ? param2.toString() : null;
    }
    
    public Boolean getParam2AsBoolean() {
        if (param2 instanceof Boolean) {
            return (Boolean) param2;
        }
        return null;
    }
}

File: src/main/node/com/smarthome/backend/model/devices/helper/DeviceListenerParams.ts
export class DeviceListenerParams {
  key?: string;
  name?: string;
  param1?: unknown;
  param2?: unknown;

  constructor(init?: Partial<DeviceListenerParams>) {
    Object.assign(this, init);
  }

  getParam1AsInt() {
    if (typeof this.param1 === "number") return Math.trunc(this.param1);
    if (typeof this.param1 === "string" && this.param1.trim()) {
      const parsed = Number(this.param1);
      return Number.isNaN(parsed) ? null : Math.trunc(parsed);
    }
    return null;
  }

  getParam1AsString() {
    return this.param1 != null ? String(this.param1) : null;
  }

  getParam1AsBoolean() {
    if (typeof this.param1 === "boolean") return this.param1;
    return null;
  }

  getParam2AsInt() {
    if (typeof this.param2 === "number") return Math.trunc(this.param2);
    if (typeof this.param2 === "string" && this.param2.trim()) {
      const parsed = Number(this.param2);
      return Number.isNaN(parsed) ? null : Math.trunc(parsed);
    }
    return null;
  }

  getParam2AsString() {
    return this.param2 != null ? String(this.param2) : null;
  }

  getParam2AsBoolean() {
    if (typeof this.param2 === "boolean") return this.param2;
    return null;
  }
}

